src/adapter/src/catalog/storage.rs

```diff
@@ -1558,13 +1558,11 @@ impl<'a> Transaction<'a> {
     }
 
     /// Upserts persisted system configuration `name` to `value`.
-    pub fn upsert_system_config(&mut self, name: &str, value: &str) -> Result<(), Error> {
+    pub fn upsert_system_config(&mut self, name: &str, value: String) -> Result<(), Error> {
```

Comment:
If this doesn't exist already, then I think we should add some tests where we persist a multi-valued system configuration and then try and read it back after a restart and make sure nothings corrupted.

Reply:
Added.